### PR TITLE
#149886251 Contribute Instructions Card 

### PR DESCRIPTION
**Pull request by andela-uofoegbu**
#### What does this PR do?
Allows users see instructions on how to make contributions to the learning map.

#### Description of Task to be completed?
- Create a card that shows how to make contributions

#### How should this be manually tested?
After cloning the repo

- Run yarn install
- Run yarn start
- Access the landing page via `localhost:3000` and click on the `LOGIN TO CONTRIBUTE` button.
- Navigate to contributions page to see card

#### Any background context you want to provide?
N/A

#### What are the relevant pivotal tracker stories?
[#149886251](https://www.pivotaltracker.com/story/show/149886251)

#### Screenshot
screen shot 2017-08-03 at 21 47 03

#### Question
N/A

### CODE REVIEW

**Comment by andela-kakpobome** [comment]
Nice! I'll merge this once I've merged Mbarak's and you align your stuff with his after rebasing.

File: src/components/instructionsCard/instructionsCard.scss [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+  height: 13.25em;
+  margin-top: 0px;
+  -webkit-animation: reversecollapse 0.5s linear;
+  animation: reversecollapse 0.5s linear;
+}
+
+@-webkit-keyframes collapse {
+  0% {
+    height: 13.25em;
+    margin-top: 0px;
```
Comment (Mbarak-Mbigo) [inline src/components/instructionsCard/instructionsCard.scss]: 0 units need not have units type.

File: src/components/instructionsCard/instructionsCard.scss [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+  }
+  100% {
+    height: 0px;
+    margin-top: -15px;
+  }
+}
+
+@keyframes collapse {
+  0% {
+    height: 13.25em;
```
Comment (Mbarak-Mbigo) [inline src/components/instructionsCard/instructionsCard.scss]: It's recommended to use `rem` over `em`

Comment (andela-uofoegbu) [inline src/components/instructionsCard/instructionsCard.scss]: Fixed this

File: src/components/instructionsCard/instructionsCard.component.tsx [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+      cardContent.classList.add('visible-content');
+    }
+  }
+
+  // tslint:disable-next-line:member-ordering
+  public render() {
+    return (
+      <Card className="card-container">
+        <CardText id="card-content">
+          <span
```
Comment (andela-kakpobome) [inline src/components/instructionsCard/instructionsCard.component.tsx]: ### Critical

Semantically, it does not make sense to use a Span (relatively low level element) to represent a header/heading. It's basically the same size as the header for the contribution card, use that same element instead.

File: src/components/instructionsCard/instructionsCard.test.tsx [hunk 4]
```diff
@@ -0,0 +1,18 @@
+import { expect } from 'chai';
+import { shallow } from 'enzyme';
+import * as React from 'react';
+import InstructionsCardComponent from './instructionsCard.component';
```
Comment (andela-kakpobome) [inline src/components/instructionsCard/instructionsCard.test.tsx]: ### Critical

There should be a linespace after lib imports and our own imports.

File: src/components/instructionsCard/instructionsCard.scss [hunk 5]
```diff
@@ -0,0 +1,135 @@
...
+    height: 13.25rem;
+    margin-top: 0rem;
+  }
+}
+
+.card-title {
+  padding-left: 1.25625rem;
+  font-size: 1rem;
+  border-bottom: 1px solid $color_iron_approx;
+  display: block;
```
Comment (andela-kakpobome) [inline src/components/instructionsCard/instructionsCard.scss]: ### Critical

Bhet why? Were you not on the call today about structuring CSS rules?

File: src/components/instructionsCard/instructionsCard.component.tsx (hunk 3 above)
Comment (andela-uofoegbu) [inline src/components/instructionsCard/instructionsCard.component.tsx]: The reason I used the span is, it doesn't contain the card heading alone, it also contains the close button.
The header for the contribution card uses a <h4>. Would it be best to wrap it with the h4 tag?

Comment (andela-kakpobome) [inline src/components/instructionsCard/instructionsCard.component.tsx]: How would the code of your new proposed structure look?

Comment (andela-uofoegbu) [inline src/components/instructionsCard/instructionsCard.component.tsx]:
```
<div
className="card-title">How to submit a contribution
  <a className="close-button" onClick={this.toggleDisplay}><i className="material-icons">close</i>
  </a>
</div>
```

Comment (andela-kakpobome) [inline src/components/instructionsCard/instructionsCard.component.tsx]: I prefer this, as the header is not some floating text, it resides in an element:

```
<div className="card-title">
  <h4>How to submit a contribution</a>
  <a className="close-button" onClick={this.toggleDisplay}>
    <i className="material-icons">close</i>
  </a>
</div>
```

Comment (andela-uofoegbu) [inline src/components/instructionsCard/instructionsCard.component.tsx]: @andela-kakpobome Fixed this!...

File: src/components/instructionCard/instructionCard.scss [hunk 6]
```diff
@@ -0,0 +1,149 @@
...
+    padding-left: 0;
+    padding-bottom: 0;
+    padding-top: 0;
+
+    .close-button {
+      position: relative;
+      margin-right: 1rem;
+      margin-top: -2.5rem;
+      vertical-align: middle;
+
```
Comment (andela-kakpobome) [inline src/components/instructionCard/instructionCard.scss]: ### Critical

Lol, you don't need to space out the rules like this, simply know and adhere to the order, please remove these spaces(no space between the rules themselves, but maintain the spacing between class definitions I mean)

File: src/components/instructionCard/instructionCard.scss [hunk 7]
```diff
@@ -0,0 +1,149 @@
...
+        text-align: center;
+        vertical-align: middle;
+      }
+
+      .upvote-icon {
+        font-size: 1rem;
+
+        color: map_get($grey, 500);
+        background: map_get($grey, 200);
```
Comment (andela-kakpobome) [inline src/components/instructionCard/instructionCard.scss]: ### Critical

Background should come first when there's no `display` or `position`

**Comment by andela-uofoegbu** [comment]
@andela-kakpobome Done.

**Pull request merged**